enterprise/admin/views.py

```diff
@@ -291,132 +292,396 @@ def _handle_bulk_upload(cls, enterprise_customer, manage_learners_form, request,
-                            user=user,
-                            enrolled_in={
-                                'name': course_name,
-                                'url': course_url,
-                                'type': 'course',
-                                'start': parse_lms_api_datetime(course_start),
-                            },
-                            enterprise_customer=enterprise_customer,
-                            email_connection=email_conn,
-                        )
-        enrolled_count = len(enrolled)
-        if enrolled_count:
-            messages.success(request, ungettext(
-                "{enrolled_count} user was enrolled to {course_id}.",
-                "{enrolled_count} users were enrolled to {course_id}.",
-                enrolled_count,
-            ).format(enrolled_count=enrolled_count, course_id=course_id))
-        if non_existing:
-            messages.warning(request, _(
-                "The following users do not have an account on {}. They have not been enrolled in the course."
-                " When these users create an account, they will be enrolled in the course automatically: {}"
-            ).format(settings.PLATFORM_NAME, ", ".join(non_existing)))
-            for email in non_existing:
-                pending_user = PendingEnterpriseCustomerUser.objects.get(
+    @classmethod
+    def program_notify(cls, enterprise_customer, program_details, users):
+        """
+        Notify learners about a program in which they've been enrolled.
+
+        Args:
+            enterprise_customer: The EnterpriseCustomer being linked to
+            program_details: Details about the specific program the learners were enrolled in
+            users: An iterable of the users or pending users who were enrolled
+        """
+        program_name = program_details.get('title')
+        program_branding = program_details.get('type')
+        program_url = program_details.get('marketing_url')
+        program_type = 'program'
+        program_start = get_earliest_start_date_from_program(program_details)
+
+        with mail.get_connection() as email_conn:
+            for user in users:
+                send_email_notification_message(
+                    user=user,
+                    enrolled_in={
+                        'name': program_name,
+                        'url': program_url,
+                        'type': program_type,
+                        'start': program_start,
+                        'branding': program_branding,
+                    },
                     enterprise_customer=enterprise_customer,
-                    user_email=email
+                    email_connection=email_conn
                 )
-                PendingEnrollment.objects.update_or_create(
-                    user=pending_user,
+
+    @classmethod
+    def get_success_message(cls, users, enrolled_in):
+        """
+        Create messages for the users who were enrolled in a course or program.
```

Comment:
Gotcha. Yes, I'll change the docstrings to reflect the singular nature of the messages. Might change the method names as well.